pkg/healthchecker/health_checker.go

```diff
@@ -65,32 +65,28 @@ func NewHealthChecker(hco *options.HealthCheckerOptions) (types.HealthChecker, e
 // CheckHealth checks for the health of the component and tries to repair if enabled.
 // Returns true if healthy, false otherwise.
 func (hc *healthChecker) CheckHealth() (bool, error) {
-	var logStartTime string
 	healthy, err := hc.healthCheckFunc()
 	if err != nil {
 		return healthy, err
 	}
-	uptime, err := hc.uptimeFunc()
-	if err != nil {
-		glog.Warningf("Failed to get the uptime: %+v", err)
-		return true, err
-	}
-	if hc.loopBackTime > 0 && uptime > hc.loopBackTime {
-		logStartTime = time.Now().Add(-hc.loopBackTime).Format(types.LogParsingTimeLayout)
-	} else {
-		logStartTime = time.Now().Add(-uptime).Format(types.LogParsingTimeLayout)
-	}
-	logPatternHealthy, err := logPatternHealthCheck(hc.service, logStartTime, hc.logPatternsToCheck)
+	logPatternHealthy, err := logPatternHealthCheck(hc.service, hc.loopBackTime, hc.logPatternsToCheck)
 	if err != nil {
 		return logPatternHealthy, err
 	}
 	if healthy && logPatternHealthy {
 		return true, nil
 	}
+	glog.Infof("healthy boolean: %v\n", healthy)
+	glog.Infof("logPatternHealthy boolean: %v\n", logPatternHealthy)
+
 	// The service is unhealthy.
 	// Attempt repair based on flag.
 	if hc.enableRepair {
 		// repair if the service has been up for the cool down period.
+		uptime, err := hc.uptimeFunc()
+		if err != nil {
```

Comment:
We shouldn't continue executing if this failed ?

Reply:
it will eventually still return false. It will output another log that may not be super necessary, but yes, it is something we can do to return early.

Reply:
I believe we should return then early. having err here means that uptime might not have a proper value.